### ISSUE TITLE
Predictive Output

### ISSUE THREAD

**Issue by danielkelshaw** [danielkelshaw/GPyBO#26]
At the moment the `predictive_posterior` function outputs a tuple of two tensors, one for the mean and the other for the covariance. These are not the easiest to interpret and you end up having to `.detach()` and `.numpy()` to get the data into a form you want a lot of the time.

It could be beneficial to provide a class which handles this sort of output more effectively and makes life easier for the user.

**Comment by danielkelshaw** [comment]
It is worth considering how the user will have to interact with the `GP` class in order to extract results about the model. The current `predictive_posterior()` method outputs a `Tuple[Tensor, Tensor]` which is the mean and covariance for the calculated Gaussian Process.

It would be useful if the user could extract the standard deviation (diagonal) elements with ease and perhaps also avoid any intermediate steps which make plotting the results more challenging.
